cairo-libs/packages/cometbft/src/types.cairo

```diff
@@ -325,20 +325,15 @@ pub impl PublicKeyImpl of PublicKeyTrait {
                 assert(signature.len() == 64, CometErrors::INVALID_SIGNATURE_LENGTH);
                 assert(pk.len() == 32, CometErrors::INVALID_PUBKEY_LENGTH);
 
-                let r_sign = signature
+                let r_sign: u256 = signature
                     .slice(0, 32)
                     .try_into()
                     .unwrap(); // Never fails as length is 32.
-                let s_sign = signature
+                let s_sign: u256 = signature
                     .slice(32, 32)
                     .try_into()
                     .unwrap(); // Never fails as length is 32.
-                let pubkey = pk.span().try_into().unwrap(); // Never fails as length is 32.
-
-                assert(
-                    verify_signature(msg, array![r_sign, s_sign].span(), pubkey),
-                    CometErrors::INVALID_ED25519_SIGNATURE,
-                );
```

Comment:
um. you forgot this ? 😅

Reply:
Wait, was accident. [b5f3f21](https://github.com/informalsystems/ibc-starknet/pull/361/commits/b5f3f21c49957c9aaa78f66db8fe10f7ab6a65df)